pkg/service/secret.go

```diff
@@ -0,0 +1,98 @@
+package service
+
+import (
+	"github.com/appuio/seiso/cfg"
+	"github.com/appuio/seiso/pkg/kubernetes"
+	"github.com/appuio/seiso/pkg/openshift"
+	log "github.com/sirupsen/logrus"
+	"github.com/thoas/go-funk"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	core "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+type SecretServiceInt interface {
+	ListNamesAndLabels() (resourceNames, labels []string, err error)
+	List(listOptions metav1.ListOptions) (resources []cfg.KubernetesResource, err error)
+	GetUnused(namespace string, resources []cfg.KubernetesResource) (unusedResources []cfg.KubernetesResource, funcErr error)
+	Delete(resources []cfg.KubernetesResource)
+}
+
+type SecretService struct {
+	Secret core.SecretInterface
+	Helper kubernetes.Kubernetes
+}
+
+func (ss SecretService) ListNamesAndLabels() (resourceNames, labels []string, err error) {
+	secrets, err := ss.List(metav1.ListOptions{})
+	secretNames, labels := GetNamesAndLabels(secrets)
+	return secretNames, labels, nil
+}
+
+// ListSecrets returns a list of secrets from a namespace
+func (ss SecretService) List(listOptions metav1.ListOptions) (resources []cfg.KubernetesResource, err error) {
```

Comment:
I would return `[]v1.Secret` instead of an Interface. Secrets are already data types with no logic, I see no reason to introduce the interface "KubernetesResource"

Reply:
well instead of duplicating the code with ConfigMaps we could reuse this service